src/main/java/com/sucy/skill/dynamic/target/TargetComponent.java

```diff
@@ -95,6 +96,7 @@ List<LivingEntity> determineTargets(final LivingEntity caster,
 
             for (LivingEntity entity : found) {
                 if (count >= max) break;
+                if (invulnerable || entity.isInvulnerable()) continue;
```

Comment:
I would think that this would be better placed in the `isValidTarget` method as a means of short-circuiting the logic there with an early return. Additionally, this logic feels incomplete.

If `invulnerable` is true, meaning that we should target invulnerable players... we skip all potential targets. If invulnerable is false and the attempted target is invulnerable, we fulfill the second half of the condition and the player is skipped, as we would expect.